lib/active_merchant/billing/gateways/borgun.rb

```diff
@@ -55,10 +55,11 @@ def refund(money, authorization, options={})
 
       def void(authorization, options={})
         post = {}
-        # TransType and TrAmount must match original values from auth or purchase.
-        _, _, _, _, _, transtype, tramount = split_authorization(authorization)
+        # TransType, TrAmount, and currency must match original values from auth or purchase.
+        _, _, _, _, _, transtype, tramount, currency = split_authorization(authorization)
         post[:TransType] = transtype
-        add_invoice(post, tramount.to_i, options)
+        post[:TrAmount] = tramount.to_i
+        post[:TrCurrency] = currency
```

Comment:
I'm going with a different approach here; I realized that this change won't be backwards compatible.